Sources/NIO/Selector.swift

```diff
@@ -406,22 +535,41 @@ final class Selector<R: Registration> {
 
         for i in 0..<ready {
             let ev = events[i]
+            guard Int32(ev.flags) & EV_ERROR == 0 else {
+                throw IOError(errnoCode: Int32(ev.data), reason: "kevent returned with EV_ERROR set")
+            }
+            var selectorEvent: SelectorEventSet = ._none
             switch Int32(ev.filter) {
             case EVFILT_USER:
                 // woken-up by the user, just ignore
-                break
+                continue
             case EVFILT_READ:
-                if let registration = registrations[Int(ev.ident)] {
-                    try body((SelectorEvent(io: .read, registration: registration)))
+                selectorEvent.formUnion(.read)
+                fallthrough // falling through here as `EVFILT_READ` also delivers `EV_EOF` (meaning `.readEOF`) and `ev.fflags != 0` (meaning `.reset`)
+            case EVFILT_EXCEPT:
+                if Int32(ev.flags) & EV_EOF != 0 {
+                    selectorEvent.formUnion(.readEOF)
+                }
+                if ev.fflags != 0 {
```

Comment:
Is this a bit over broad?

Reply:
don't think so, AFAIK matches `EPOLLERR`